src/torrent.cpp

```diff
@@ -9515,8 +9492,8 @@ namespace libtorrent
 		// now, iterate over all time critical pieces, in order of importance, and
 		// request them from the peers, in order of responsiveness. i.e. request
 		// the most time critical pieces from the fastest peers.
-		for (std::vector<time_critical_piece>::iterator i = m_time_critical_pieces.begin()
-			, end(m_time_critical_pieces.end()); i != end; ++i)
+		bool non_first_piece{false};
```

Comment:
this is a bit of a nit-pick, but would you mind negating the name and meaning of this? It just feels like a double-negation right now